File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int n=nums.size();
        int i=0;
        while(i<n){
            int correctindex=nums[i]-1;
            if (nums[i]!=nums[correctindex]){
                int temp=nums[i];
                nums[i]=nums[correctindex];
                nums[correctindex]=temp;
            }
            else{i++;}
        }
        vector<int>ans;
        for(int j=0;j<n;j++){
            if(nums[j]!=j+1){
                ans.push_back(j+1);
            }

        }
        return ans;
    }
};

File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        n = len(nums)
        i = 0

        # Cyclic sort
        while i < n:
            correct_index = nums[i] - 1

            if nums[i] == nums[correct_index]:
                i += 1
            else:
                nums[i], nums[correct_index] = nums[correct_index], nums[i]

        # Find missing numbers
        ans = []
        for j in range(n):
            if nums[j] != j + 1:
                ans.append(j + 1)

        return ans